DataDog/serverless-plugin-datadog
# Allow bypassing the site check when in testingMode

### What does this PR do?

- Rename `integrationTesting` to `testingMode`
- Bypass site check when `testingMode` is enabled
- Remove `integrationTesting` from README.md since it will not be used by users
- Create a relevant test for bypassing the site check

### Motivation

We needed a way for developers to disable the site check so we can set it to another site for testing purposes.

### Testing Guidelines

I created a new test in `index.spec.ts` that checks that an error will not be thrown when using `testingMode=true` and `site` is not a valid site found in `siteList`.

I then manually tested by building the tool locally to ensure that the plugin succeeds and does not throw an error when `testingMode=true` and `site` is not in `siteList`. When `testingMode=false` or wasn't in the config, an error was successfully thrown when the `site` was not in `siteList`.

### Additional Notes

There is no need for `testingMode` to appear in `README.md` since it will not be used by users. Therefore, I did not include it and I removed the original `integrationTesting` row in the table in `README.md`.

### Types of changes


### Check all that apply
